src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURE = 5;
+    private static final int MAX_LENGTH = 101;
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(MAX_FIGURE)) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_LENGTH));
+            case 1:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_LENGTH),
+                        random.nextInt(MAX_LENGTH),
+                        random.nextInt(MAX_LENGTH));
+            case 2:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_LENGTH),
+                        random.nextInt(MAX_LENGTH));
+            case 3:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_LENGTH),
+                        random.nextInt(MAX_LENGTH));
+            case 4:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_LENGTH));
+            default:
+                return getDefaultFigure();
```

Comment:
1) This part has a problem with code duplication. You generate random sizes using one formula. Let's imagine tomorrow you'll need to change this formula, then you'll have to go over each line and make a change. It's insufficient. In this case, you can create a separate method for this computation and reuse it.
2) Since each figure needs a color you can extract colorSupplier.getRandomColor() to a separate variable and reuse